plugins/modules/azure_rm_virtualmachine.py

```diff
@@ -1813,6 +1896,47 @@ def exec_module(self, **kwargs):
                     if self.license_type is not None:
                         vm_resource.license_type = self.license_type
 
+                    if self.vm_identity is not None:
+                        # If 'append' is set to True save current user assigned managed identities to use later
+                        if self.vm_identity.get('user_assigned_identities',{}) is not None and self.vm_identity.get('user_assigned_identities',{}).get('append', False) is True:
+                            if 'identity' in vm_dict and 'userAssignedIdentities' in vm_dict['identity']:
+                                current_user_assigned_identities_dict = { uami:dict() for uami in vm_dict['identity']['userAssignedIdentities'].keys() }
+                                vm_identity_user_assigned_append = True
+                            else:
+                                # Nothing to append to
+                                vm_identity_user_assigned_append = False
+                        else:
+                            # 'append' is False or unset
+                            vm_identity_user_assigned_append = False
+                        # If there are identities in 'id' and 'UserAssigned' in type
+                        if 'UserAssigned' in self.vm_identity.get('type') and len(self.vm_identity.get('user_assigned_identities',{}).get('id',[])) != 0:
+                            user_assigned_identities_dict = { uami:dict() for uami in self.vm_identity.get('user_assigned_identities').get('id') }
+                            # If there are identities to append, merge the dicts
+                            if vm_identity_user_assigned_append:
+                                user_assigned_identities_dict = { **user_assigned_identities_dict, **current_user_assigned_identities_dict}
+                            # Save the identity
+                            vm_resource.identity = self.compute_models.VirtualMachineIdentity(
+                                type=self.vm_identity.get('type'),
+                                user_assigned_identities=user_assigned_identities_dict
+                            )
+                        # If there are no identities in 'id' and 'UserAssigned' in type
+                        elif 'UserAssigned' in self.vm_identity.get('type') and len(self.vm_identity.get('user_assigned_identities',{}).get('id',[])) == 0:
+                            # Fail if append is False
+                            if vm_identity_user_assigned_append is False:
+                                self.fail("UserAssigned specified but no User Assigned IDs provided and no UserAssigned identities are currently assigned to the VM")
```

Comment:
Too long (165 characters)
```suggestion
                                self.fail("UserAssigned specified but no User Assigned IDs provided and no UserAssigned identities are currently assigned to the VM")
```

Reply:
I have divided this line in two. Do you need me to rephrase the error message? If code reaches this point, the module arguments specify that a User Assigned Managed Identity has to be appended to the VM, but there aren't identities in the module arguments nor in the current VM configuration